ProjectObsidian/ProtoFlux/Audio/SineGenerator.cs

```diff
@@ -29,22 +29,16 @@ public class SineGeneratorProxy : ProtoFluxEngineProxy, IAudioSource
 
         public int ChannelCount => 1;
 
-        protected override void OnAwake()
-        {
-            base.OnAwake();
-            Frequency = 440f;
-            Amplitude = 1f;
-            Phase = 0f;
-        }
-
+        // TODO: Make this not advance time on each read
```

Comment:
Decouple time progression from the Read method by introducing a shared, externally managed time state.
?? // just a suggestion

Reply:
Noted by Nytra this was attempted but caused issues with audio quality.